### ISSUE TITLE
[WINDOWS] HazelcastOSGiIntegrationTest.serviceRetrievedSuccessfully

### ISSUE THREAD

**Issue by hasancelik** [hazelcast/hazelcast#9892]
```
java.lang.ClassNotFoundException: com.hazelcast.osgi.HazelcastOSGiService not found by PAXEXAM-PROBE-2e5a8c6d-b091-47a7-b337-eddddbc57e12 [17]
	at org.apache.felix.framework.BundleWiringImpl.findClassOrResourceByDelegation(BundleWiringImpl.java:1556)
	at org.apache.felix.framework.BundleWiringImpl.access$400(BundleWiringImpl.java:77)
	at org.apache.felix.framework.BundleWiringImpl$BundleClassLoader.loadClass(BundleWiringImpl.java:1993)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:358)
	at java.lang.Class.getDeclaredMethods0(Native Method)
	at java.lang.Class.privateGetDeclaredMethods(Class.java:2625)
	at java.lang.Class.getDeclaredMethods(Class.java:1868)
	at org.junit.runners.model.TestClass.<init>(TestClass.java:41)
	at org.junit.runners.ParentRunner.<init>(ParentRunner.java:73)
	at org.junit.runners.BlockJUnit4ClassRunner.<init>(BlockJUnit4ClassRunner.java:55)
	at org.ops4j.pax.exam.invoker.junit.internal.ContainerTestRunner.<init>(ContainerTestRunner.java:53)
	at org.ops4j.pax.exam.invoker.junit.internal.ContainerTestRunnerBuilder.runnerForClass(ContainerTestRunnerBuilder.java:46)
	at org.junit.runners.model.RunnerBuilder.safeRunnerForClass(RunnerBuilder.java:57)
	at org.ops4j.pax.exam.invoker.junit.internal.ContainerTestRunnerClassRequest.getRunner(ContainerTestRunnerClassRequest.java:60)
	at org.junit.internal.requests.FilterRequest.getRunner(FilterRequest.java:33)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:136)
	at org.ops4j.pax.exam.invoker.junit.internal.JUnitProbeInvoker.invokeViaJUnit(JUnitProbeInvoker.java:124)
	at org.ops4j.pax.exam.invoker.junit.internal.JUnitProbeInvoker.findAndInvoke(JUnitProbeInvoker.java:94)
	at org.ops4j.pax.exam.invoker.junit.internal.JUnitProbeInvoker.call(JUnitProbeInvoker.java:77)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.ops4j.pax.exam.rbc.internal.RemoteBundleContextImpl.remoteCall(RemoteBundleContextImpl.java:86)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at sun.rmi.server.UnicastServerRef.dispatch(UnicastServerRef.java:322)
	at sun.rmi.transport.Transport$2.run(Transport.java:202)
	at sun.rmi.transport.Transport$2.run(Transport.java:199)
	at java.security.AccessController.doPrivileged(Native Method)
	at sun.rmi.transport.Transport.serviceCall(Transport.java:198)
	at sun.rmi.transport.tcp.TCPTransport.handleMessages(TCPTransport.java:567)
	at sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.run0(TCPTransport.java:828)
	at sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.access$400(TCPTransport.java:619)
	at sun.rmi.transport.tcp.TCPTransport$ConnectionHandler$1.run(TCPTransport.java:684)
	at sun.rmi.transport.tcp.TCPTransport$ConnectionHandler$1.run(TCPTransport.java:681)
	at java.security.AccessController.doPrivileged(Native Method)
	at sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.run(TCPTransport.java:681)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)
```
https://hazelcast-l337.ci.cloudbees.com/view/Hazelcast-Windows/job/Hazelcast-3.x-Windows/com.hazelcast$hazelcast/11/testReport/junit/com.hazelcast.osgi/HazelcastOSGiIntegrationTest/serviceRetrievedSuccessfully_com_hazelcast_osgi_HazelcastOSGiIntegrationTest_serviceRetrievedSuccessfully_PaxRunnerTestContainer_felix_/

**Comment by Donnerbart** [comment]
This test is also failing under Linux for me in IntelliJ. The comment from Jaromir regarding the wrong module directory doesn't help me:
```
        // Is this test failing in your IDE?
        // Some versions of Intellij IDEA use a wrong working directory in multi-module
        // Maven projects. See this for a fix: https://youtrack.jetbrains.com/issue/IDEA-60965
```
I've set the module directory to different folders, nothing helps to get this test green. So I have no idea how to improve this.

@jerrinot Any idea how to debug this?

**Comment by tombujok** [comment]
@Donnerbart are you working on this one and planning to fix in 3.8.2?

**Comment by mesutcelik** [comment]
you can also reproduce it locally if you mess up with the `url` in the HazelcastOSGiIntegrationTest

**Comment by Donnerbart** [comment]
For me the test never worked locally, without changing anything. At the moment it's running fine on the `master` branch, but I have no clue why. I just reported it here, since the issue is labeled with `[WINDOWS]`, but the test was also failing on my Linux system. I was never working on this, just reporting failures from my local test runs :)

**Comment by mesutcelik** [comment]
I was directing my comment to @hasancelik as he is the assigned one :) no worries...

**Comment by Donnerbart** [comment]
Sure, I just saw Tom's comment from May. Also this test failed for a very, very long time on my machine, so it's good to see that it's finally working (at least with Linux).

I also have a pending 3.10 issue for explicit Windows test failures (https://github.com/hazelcast/hazelcast-enterprise/issues/1323). We already fixed a bunch of them, but some may still be broken. Would be interesting if this test is (still) one of them.
